src/realm/db.cpp

```diff
@@ -2403,11 +2406,33 @@ bool DB::call_with_lock(const std::string& realm_path, CallbackWithLock callback
     return false;
 }
 
-std::vector<std::pair<std::string, bool>> DB::get_core_files(const std::string& realm_path)
+std::vector<std::pair<std::string, bool>> DB::get_core_files(const std::string& realm_path, uint64_t core_file_type)
 {
     std::vector<std::pair<std::string, bool>> files;
-    files.emplace_back(std::make_pair(realm_path, false));
-    files.emplace_back(std::make_pair(realm_path + ".management", true));
+    if ((DB::CoreFileType::Lock & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".lock", false));
+    }
+    if ((DB::CoreFileType::Storage & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path, false));
+    }
+    if ((DB::CoreFileType::Management & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".management", true));
+    }
+    if ((DB::CoreFileType::Note & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".note", false));
+    }
+    if ((DB::CoreFileType::Log & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".log", false));
+    }
+    if ((DB::CoreFileType::LogA & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".log_a", false));
+    }
+    if ((DB::CoreFileType::LogB & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".log_b", false));
+    }
+    if ((DB::CoreFileType::Backup & core_file_type)) {
+        files.emplace_back(std::make_pair(realm_path + ".realm.backup", false));
```

Comment:
I have removed it with 85e5bf504. You're probably right, backup is a bit different. It would have been nice to have everything in one place but maybe we come up with a better idea in the future.